tests/5.1/allocate/test_calloc_host.c

```diff
@@ -0,0 +1,76 @@
+//===------------------------ test_calloc_host.c --------------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// Tests the omp_calloc routine to make sure memory is properly allocated
+// on the host and is zero initialized. Allocates an array and makes sure
+// each value is set to 0 to start. Then adds i to each index and validates the
+// changes persisted and were written correctly.
+//
+//===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_omp_calloc_host() {
+  int errors = 0;
+  int *x;
+
+  x = (int *)omp_calloc(64, N*sizeof(int), omp_default_mem_alloc);
+
+  if (x == NULL) {
+    OMPVV_ERROR("omp_calloc returned null"); 
+    return (1); 
+  }
+
+  int not_init_to_zero = 0;
+  int not_correct_updated_values = 0;
+
+  #pragma omp parallel for
+  for (int i = 0; i < N; i++) {
+    if (x[i] != 0) {
+      #pragma omp atomic
```

Comment:
When compiled with LLVM. `#pragma omp atomic` gives the following error messages:
```
tests/5.1/allocate/test_calloc_host.c:38:26: error: the statement for 'atomic' must be an expression statement of form '++x;', '--x;', 'x++;', 'x--;', 'x binop= expr;', 'x = x binop expr' or 'x = expr binop x', where x is an lvalue expression with scalar type
      not_init_to_zero = 1;
                         ^
tests/5.1/allocate/test_calloc_host.c:38:26: note: expected built-in binary operator
      not_init_to_zero = 1;
                         ^
tests/5.1/allocate/test_calloc_host.c:51:36: error: the statement for 'atomic' must be an expression statement of form '++x;', '--x;', 'x++;', 'x--;', 'x binop= expr;', 'x = x binop expr' or 'x = expr binop x', where x is an lvalue expression with scalar type
      not_correct_updated_values = 1;
                                   ^
tests/5.1/allocate/test_calloc_host.c:51:36: note: expected built-in binary operator
      not_correct_updated_values = 1;
                                   ^
2 errors generated.
```
This can be solved by adding `#pragma omp atomic write`.

Reply:
Yeah awesome catch, I'll patch it up. Thanks @fel-cab!